packages/ember-metal/lib/property_set.js

```diff
@@ -55,7 +55,8 @@ export function set(obj, keyName, value, tolerant) {
     return setPath(obj, keyName, value, tolerant);
   }
 
-  assert('calling set on destroyed object', !obj.isDestroyed);
+  assert(`calling set on destroyed object: ${obj.constructor ? obj : null}.${keyName} = ${value}`,
```

Comment:
Does a `.toString` check satisfy the same edge cases with EmptyObject that this `obj.constructor` check does?  If so, I'd rather use it to make it clearer what we need.

`${obj.toString ? obj.toString() : ''}` or something?